sympy/functions/elementary/integers.py

```diff
@@ -134,10 +134,9 @@ def _eval_number(cls, arg):
     def _eval_as_leading_term(self, x, cdir=0):
         arg = self.args[0].as_leading_term(x)
         arg0 = arg.subs(x, 0)
-        if arg0.is_integer:
-            return arg0
-        elif arg0.is_finite:
-            return self.func(arg0)
+        if arg0.is_finite:
+            res = self._eval_nseries(x, n=1, logx=None, cdir=cdir)
+            return res.as_leading_term(x, cdir=cdir)
```

Comment:
This could probably be `res` as `floor._eval_nseries` only returns a single term.